### PR TITLE
lz4: conan v2 support

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
`project()` before `cmake_minimum_required()` in their CMakeLists, I'll push a patch.

**Review by uilianries — APPROVED** [review]

**Comment by SpaceIm** [comment]
patch submitted upstream and merged: https://github.com/lz4/lz4/pull/1133

And actually I need to backport this full patch, otherwise options values are not honored since they are passed through toolchain (ie they must be moved after `project()`)

**Review by uilianries — DISMISSED** [review]

File: recipes/lz4/all/conandata.yml [hunk 1]
```diff
@@ -6,8 +6,9 @@ sources:
     sha256: 658ba6191fa44c92280d4aa2c271b0f4fbc0e34d249578dd05e50e76d0e5efcc
     url: https://github.com/lz4/lz4/archive/v1.9.2.tar.gz
 patches:
+  "1.9.3":
+    - patch_file: "patches/0003-cmake-minimum-required-first-1.9.3.patch"
```
Comment (prince-chrismc) [inline recipes/lz4/all/conandata.yml:10]:
```suggestion
    - patch_file: "patches/0003-cmake-minimum-required-first-1.9.3.patch"
       patch_type: "backport"
       patch_source: "https://github.com/lz4/lz4/pull/1133"
```

File: recipes/lz4/all/conandata.yml [hunk 2]
```diff
@@ -6,8 +6,9 @@ sources:
...
   "1.9.2":
     - patch_file: "patches/0001-cmake-add-shared-DEFINE_SYMBOL.patch"
-      base_path: "source_subfolder"
     - patch_file: "patches/0002-cmake-optional-cli.patch"
-      base_path: "source_subfolder"
+    - patch_file: "patches/0003-cmake-minimum-required-first-1.9.2.patch"
```
Comment (prince-chrismc) [inline recipes/lz4/all/conandata.yml:14]:
```suggestion
    - patch_file: "patches/0003-cmake-minimum-required-first-1.9.2.patch"
       patch_type: "backport"
       patch_source: "https://github.com/lz4/lz4/pull/1133"
```

**Review by prince-chrismc — COMMENTED** [review]
I would really like to encourage everyone to start added the patch metadata 🙏 

https://github.com/conan-io/conan-center-index/blob/master/docs/conandata_yml_format.md#patches-fields

Thanks for opening a PR upstream

**Review by uilianries — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Pull request merged**